packages/concerto-core/lib/decoratormanager.js

```diff
@@ -474,6 +654,38 @@ class DecoratorManager {
             this.applyDecorator(property, type, decorator);
         }
     }
+    /**
+     * extracts a Command from a ClassDeclaration or its properties, as required.
+     * @param {String} namespace the namespace for the declaration
+     * @param {*} declaration the class declaration
+     * @param {Object} decoratorDict the dictionary with extracted decorators
+     * @param {boolean} shouldRemoveDecorators whether to remove decorators from base model
+     */
+    static extractDecoratorFromModel(namespace, declaration, decoratorDict, shouldRemoveDecorators) {
```

Comment:
This function appears to have side-effects:
1. To modify the declaration
2. To update the decoratorDict

Is there a way to make this more functional?

Reply:
updated in latest commits.